R/bundle.R

```diff
@@ -270,9 +270,37 @@ writeBundle <- function(bundleDir, bundlePath, verbose = FALSE) {
 
   tarImplementation <- Sys.getenv("RSCONNECT_TAR", "internal")
   logger(sprintf("Using tar: %s", tarImplementation))
+
+  if (tarImplementation == "internal") {
+    detectLongNames(bundleDir)
+  }
+
   utils::tar(bundlePath, files = NULL, compression = "gzip", tar = tarImplementation)
 }
 
+# Scan the bundle directory looking for long user/group names.
+#
+# Warn that the internal tar implementation may produce invalid archives.
+# https://github.com/rstudio/rsconnect/issues/446
+# https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17871
+detectLongNames <- function(bundleDir, lengthLimit = 32) {
+  files <- list.files(bundleDir, recursive = TRUE, all.files = TRUE,
+                      include.dirs = TRUE, no.. = TRUE, full.names = FALSE)
+  for (f in files) {
+    info <- file.info(file.path(bundleDir,f))
+    cat(sprintf("f=%s; u=%s (%d); g=%s, (%d)\n", f, info$uname, nchar(info$uname), info$grname, nchar(info$grname)))
```

Comment:
Do we always want to emit this line, or only if we detect a long user / group name?

Reply:
No. This was a mistake. It's gone.